### PR TITLE
[MFTF] Cancel created orders after checks on frontend

### LINKED ISSUE #28324: [Issue] [MFTF] Cancel created orders after checks on frontend
This issue is automatically created based on existing pull request: magento/magento2#27698: [MFTF] Cancel created orders after checks on frontend

---------
### Description
According to the isolation guidelines of Magento Functional Test Framework, all data created during tests have to be deleted. As we can't delete created orders - there's no way to delete an order neither via backend or webapi - I suppose it'd be an good idea to cancel them.

https://devdocs.magento.com/mftf/docs/guides/test-isolation.html

Updated `CheckCheckoutSuccessPageAsRegisterCustomerTest` and `CheckCheckoutSuccessPageAsGuestTest` after blocks, to cancel all orders which have been created in these tests.

### New Action Groups

1. `AdminGridBulkActionGroup` - massive action for all rows on Admin Grid page
2. `AdminGridColumnShowActionGroup` - shows new column on Admin Grid page

### Questions or comments

I spent so much time, trying to cancel orders by ids via webapi - but still don't know how to do that in MFTF. If someone has any thoughts or ideas - they'd be appreciated.

### PR DESCRIPTION
**Pull request by elevinskii**
### Description
According to the isolation guidelines of Magento Functional Test Framework, all data created during tests have to be deleted. As we can't delete created orders - there's no way to delete an order neither via backend or webapi - I suppose it'd be an good idea to cancel them.

https://devdocs.magento.com/mftf/docs/guides/test-isolation.html

Updated `CheckCheckoutSuccessPageAsRegisterCustomerTest` and `CheckCheckoutSuccessPageAsGuestTest` after blocks, to cancel all orders which have been created in these tests.

### New Action Groups

1. `AdminGridBulkActionGroup` - massive action for all rows on Admin Grid page
2. `AdminGridColumnShowActionGroup` - shows new column on Admin Grid page

### Questions or comments

I spent so much time, trying to cancel orders by ids via webapi - but still don't know how to do that in MFTF. If someone has any thoughts or ideas - they'd be appreciated.

### Resolved issues:
1. [x] resolves magento/magento2#28324: [MFTF] Cancel created orders after checks on frontend

### CODE REVIEW

File: app/code/Magento/Checkout/Test/Mftf/Test/CheckCheckoutSuccessPageTest.xml [hunk 1]
```diff
@@ -27,10 +27,26 @@
         </before>
 
         <after>
+            <!--Cancel orders-->
+            <actionGroup ref="AdminLoginActionGroup" stepKey="adminLogin"/>
+            <actionGroup ref="AdminOrdersGridClearFiltersActionGroup" stepKey="goToOrdersPage"/>
+            <waitForLoadingMaskToDisappear stepKey="waitAfterClearFilters"/>
```
Comment (lbajsarowicz) [inline app/code/Magento/Checkout/Test/Mftf/Test/CheckCheckoutSuccessPageTest.xml]: If there's no such action - you can add it to `AdminOrdersGridClearFiltersActionGroup`

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
Could you - by the way - extract the Tests to separate files? (There are 2 tests in the same file)

--- EDIT ---
I see that you used outdated `2.4-develop`, that's why you have 2 tests in same file. Now these tests were isolated.

**Comment by elevinskii** [comment]
> I see that you used outdated `2.4-develop`, that's why you have 2 tests in same file. Now these tests were isolated.

Sorry, overlooked that. Now the branch is updated.

File: app/code/Magento/Checkout/Test/Mftf/Test/CheckCheckoutSuccessPageTest.xml (hunk 1 above)
Comment (elevinskii) [reply app/code/Magento/Checkout/Test/Mftf/Test/CheckCheckoutSuccessPageTest.xml]: That's done - I've added wait action in the end of `AdminOrdersGridClearFiltersActionGroup` and did small refactoring for tests where this action group is used.

File: app/code/Magento/Checkout/Test/Mftf/Test/CheckCheckoutSuccessPageTest/CheckCheckoutSuccessPageAsGuestTest.xml [hunk 2]
```diff
@@ -23,7 +23,22 @@
...
+                <argument name="columnLabel" value="Customer Email"/>
+            </actionGroup>
+            <actionGroup ref="AdminGridFilterFillInputFieldActionGroup" stepKey="filterOrdersByCustomerEmail">
+                <argument name="filterInputName" value="customer_email"/>
+                <argument name="filterValue" value="{{CustomerEntityOne.email}}"/>
+            </actionGroup>
+            <actionGroup ref="AdminGridFilterApplyActionGroup" stepKey="applyFilters"/>
+            <actionGroup ref="AdminGridBulkActionGroup" stepKey="cancelOrders">
+                <argument name="actionLabel" value="Cancel"/>
+            </actionGroup>
```
Comment (lbajsarowicz) [inline app/code/Magento/Checkout/Test/Mftf/Test/CheckCheckoutSuccessPageTest/CheckCheckoutSuccessPageAsGuestTest.xml:39]: Awesome! Perfect test example :exclamation:

**Review by lbajsarowicz — DISMISSED** [review]
:heavy_check_mark:  Thank you for your contribution

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-7334](https://jira.corp.magento.com/browse/ENGCOM-7334) has been created to process this Pull Request
:eight_spoked_asterisk: @lbajsarowicz, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Comment by engcom-Echo** [comment]
Failed functional tests not related to the changes in this PR

**Review by slavvka — CHANGES_REQUESTED** [review]
@elevinskii Functional Tests EE are constantly failing on this PR so that means it is caused by the PR. Please fix them.

**Comment by lbajsarowicz** [comment]
Failing tests belong to MSI project. These tests tend to fail randomly, I reviewed them and I'm sure that there's no reference to the changes introduced.

**Comment by slavvka** [comment]
@elevinskii @lbajsarowicz I was trying to deliver them to mainline and reran this build multiple times. All the time they failed identically. Than I restarted them here and got the same results. After I removed this PR from the batch, the tests passed.

**Comment by slavvka** [comment]
You could try to reran them here so many times and get them green. meanwhile I tend to think that they are failing due to the PR changes.

**Comment by VladimirZaets** [comment]
@magento run all tests

**Comment by slavvka** [comment]
@magento create issue

**Comment by slavvka** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by sidolov** [comment]
@magento run all tests

**Comment by sidolov** [comment]
@elevinskii please, take a look at failed test

**Comment by engcom-Echo** [comment]
Hi @elevinskii.Please, take a look at a failed tests. Can you fix them?

**Comment by gabrieldagama** [comment]
The risk was set to `low` due to: Changes on this PR don't affect the platform, just the test suite.

**Review by lbajsarowicz — APPROVED** [review]
✔️  Looks correct for me.

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-7334](https://jira.corp.magento.com/browse/ENGCOM-7334) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
QA not applicable

**Pull request merged**